lib/plugins/aws/deploy/index.js

```diff
@@ -127,6 +124,7 @@ class AwsDeploy {
       'aws:deploy:deploy:createStack': async () => this.createStack(),
 
       'aws:deploy:deploy:checkForChanges': async () => {
+        await this.ensureValidBucketExists();
         await this.setBucketName();
```

Comment:
This line I believe now can be removed

Reply:
This line is still needed, but potentially could be moved inside `ensureValidBucketExists` - after the logic of `ensureValidBucketExists`, in scenario where we create a new bucket via CF update, we don't know the bucket name and we have to retrieve it via SDK and that call handles that. What do you think, would it be better to just call it at the end of processing in `ensureValidBucketExists`?

Reply:
I propose to simplify it as:
- Remove `setBucketName()` entirely (I've checked and it's safe against plugins)
- In `ensureValidBucketExists`,ensure following logic is run:

```javascript
if (!this.bucketName) {
  this.bucketName = await this.provider.getServerlessDeploymentBucketName();
}
```

Reply:
I think we cannot remove `setBucketName` entirely, because it's also used in `rollback` and `deployList` logic. as for the proposed syntax, we can definitely do that but adjust the logic in `ensureValidBucketExists` a bit more, because it returns as soon as possible when checks are (not) satisfied

Reply:
I've removed the additional `setBucketName` in `checkForChanges` hook and adjusted the implementation of `ensureValidBucketExists` - at the end of the function we should unconditionally call that method because we know the bucketName is not set (because we just ran the update for missing bucket in CF template)